specs/decimal/decimal.md

```diff
@@ -68,125 +229,106 @@ namespace Microsoft.Windows.Foundation
-
-        void SetFromBoolean(Boolean value);
-        void SetFromInt16(Int16 value);
-        void SetFromInt32(Int32 value);
-        void SetFromInt64(Int64 value);
-        void SetFromUInt8(UInt8 value);
-        void SetFromUInt16(UInt16 value);
-        void SetFromUInt32(UInt32 value);
-        void SetFromUInt64(UInt64 value);
-        void SetFromSingle(Single value);
-        void SetFromDouble(Double value);
-        void SetFromString(String value);                           // LCID=LOCALE_INVARIANT
-        void SetFromStringWithSystemDefaultLocale(String value);    // LCID=GetSystemDefaultLCID()
-        void SetFromStringWithUserDefaultLocale(String value);      // LCID=GetUserDefaultLCID()
-        void SetFromStringWithThreadLocale(String value);           // LCID=GetThreadLocale()
-        void SetFromStringWithInvariantLocale(String value);        // LCID=LOCALE_INVARIANT
-        void Set(IInspectable value);
-        void SetFromDecimal(Decimal value);
-        void SetFromDecimalValue(DecimalValue value);
-
-        Boolean ToBoolean();
-        Int16 ToInt16();
-        Int32 ToInt32();
-        Int64 ToInt64();
-        UInt8 ToUInt8();
-        UInt16 ToUInt16();
-        UInt32 ToUInt32();
-        UInt64 ToUInt64();
-        Single ToSingle();
-        Double ToDouble();
-        //String ToString(); inherited from IStringable     // LCID=LOCALE_INVARIANT
-        String ToStringWithSystemDefaultLocale();           // LCID=GetSystemDefaultLCID()
-        String ToStringWithUserDefaultLocale();             // LCID=GetUserDefaultLCID()
-        String ToStringWithThreadLocale();                  // LCID=GetThreadLocale()
-        String ToStringWithInvariantLocale();               // LCID=LOCALE_INVARIANT
-        IInspectable ToObject();
-        Decimal ToDecimal();                //TODO: Rename to Copy(value) or Clone(value) ?
-        DecimalValue ToDecimalValue();
-
-        /// Return true if (this == value).
-        Boolean Equals(Decimal value);
-
-        /// Compare this decimal with value.
-        /// @return 0 if this and value are equal, <0 if this is less than value or >0 if this is greater than value.
-        Int32 Compare(Decimal value);
+        static DecimalValue FromBoolean(Boolean value);
+        static DecimalValue FromInt16(Int16 value);
+        static DecimalValue FromInt32(Int32 value);
+        static DecimalValue FromInt64(Int64 value);
+        static DecimalValue FromUInt8(UInt8 value);
+        static DecimalValue FromUInt16(UInt16 value);
+        static DecimalValue FromUInt32(UInt32 value);
+        static DecimalValue FromUInt64(UInt64 value);
+        static DecimalValue FromSingle(Single value);
+        static DecimalValue FromDouble(Double value);
+        static DecimalValue FromString(String value);                          // LCID=LOCALE_INVARIANT
+        static DecimalValue FromStringWithSystemDefaultLocale(String value);   // LCID=GetSystemDefaultLCID()
```

Comment:
There was heated debate in API design in 2018ish about how to represent locales in API parameters.  Our public documentation - https://learn.microsoft.com/en-us/windows/win32/intl/locale-identifiers - suggests using "locale names" (like https://learn.microsoft.com/en-us/windows/win32/intl/locale-names) as the preferred currency.  The API group landed on "it's just a string with a structure" rather than introducing some custom type.  See also https://learn.microsoft.com/en-us/uwp/api/windows.globalization.language.languagetag?view=winrt-26100 for where that peeks out.

Reply:
RECOMMEND: Per other comment DON'T expose LCID. Replace ctor and oper= with the trio of named methods

* ToString()
* ToStringInvariant()
* ToStringWithLocale(string localeName)
* 
(and the inverse From... equivalents)

Thanks for the link re local names, will be useful for the implementation

Reply:
DISCUSSION: `LOCALE_NAME_USER_DEFAULT`, `..._INVARIANT` and `...SYSTEM_DEFAULT` are defined in winnls.h

```
// String based NLS APIs
//

#define LOCALE_NAME_USER_DEFAULT            NULL
#define LOCALE_NAME_INVARIANT               L""
#define LOCALE_NAME_SYSTEM_DEFAULT          L"!x-sys-default-locale"
```

`NULL` vs `""` makes this problematic for WinRT HSTRINGs so we chose UserDefault for our default (e.g. `ToString()`) for consistency with C# (and other languages where 'current locale` / etc rules commonly apply) and an explicitly named `...Invariant` method for the invariant locale.